ios/Runner/AppDelegate.swift

```diff
@@ -23,12 +26,17 @@ let apnTokenKey = "apnToken"
         let twakeApnChannel = FlutterMethodChannel(
             name: "twake_apn",
             binaryMessenger: controller.binaryMessenger)
-        twakeApnChannel.setMethodCallHandler({
-          (call: FlutterMethodCall, result: @escaping FlutterResult) -> Void in
-            if call.method == "getToken" {
+        twakeApnChannel.setMethodCallHandler { [weak self ] call, result in
```

Comment:
just curious, why do you use `weak` here

Reply:
for swift, always use [weak self] in escaping closure to avoid memory leak
it won't increase the count in ARC
actually, AppDelegate is a singleton, so it isn't necessary
even if it's weak self it's okay